docs/prd/component-dependencies.md

```
@@ -0,0 +1,242 @@
+```go
+// IsFileDependency returns true if this is a file dependency.
+func (d ComponentDependency) IsFileDependency() bool {
+    return d.Kind == "file"
+}
+
+// IsFolderDependency returns true if this is a folder dependency.
+func (d ComponentDependency) IsFolderDependency() bool {
+    return d.Kind == "folder"
+}
+
+// IsComponentDependency returns true if this is a component dependency (not file or folder).
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)